### ISSUE TITLE
NoClassDefFoundError with Robolectric 4.7.2, sdk 30 and ContentProvider tests

### ISSUE THREAD

**Issue by ninniuz** [robolectric/robolectric#6883]
### Description
When running a test that uses `Robolectric.setupContentProvider` or `Robolectric.buildContentProvider` with Robolectric 4.7.x and `compileSdk = 30` I get a `java.lang.NoClassDefFoundError` on class `android/content/AttributionSource` no matter which sdk I set in the `@Config` annotation.

The same does not happen with:
- Robolectric < 4.7.x
- `compileSdk > 30` 

### Steps to Reproduce
- Set `compileSdk = 30`
- Add `Robolectric.setupContentProvider` in a test
- Use Robolectric 4.7.x
- Run tests

### Robolectric & Android Version
Robolectric: 4.7.2
Android compile SDK: 30

### Link to a public git repo demonstrating the problem:
https://github.com/ninniuz/robolectric-content-provider-fail

Running tests with `./gradlew :app:testDebugUnitTest` will fail with the following stacktrace

```
java.lang.NoClassDefFoundError: android/content/AttributionSource
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2473)
	at org.robolectric.internal.bytecode.ShadowWrangler.findShadowMethodDeclaredOnClass(ShadowWrangler.java:319)
	at org.robolectric.internal.bytecode.ShadowWrangler.findShadowMethod(ShadowWrangler.java:288)
	at org.robolectric.internal.bytecode.ShadowWrangler.pickShadowMethod(ShadowWrangler.java:265)
	at org.robolectric.internal.bytecode.ShadowWrangler.classInitializing(ShadowWrangler.java:160)
	at org.robolectric.internal.bytecode.RobolectricInternals.classInitializing(RobolectricInternals.java:21)
	at android.app.AppOpsManager.<clinit>(AppOpsManager.java)
	at android.app.SystemServiceRegistry$62.createService(SystemServiceRegistry.java:795)
	at android.app.SystemServiceRegistry$62.createService(SystemServiceRegistry.java:790)
	at android.app.SystemServiceRegistry$CachedServiceFetcher.getService(SystemServiceRegistry.java:1702)
	at android.app.SystemServiceRegistry.getSystemService(SystemServiceRegistry.java:1382)
	at android.app.ContextImpl.getSystemService(ContextImpl.java:1918)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.robolectric.shadows.ShadowContextImpl$_ContextImpl_$$Reflector4.getSystemService(Unknown Source)
	at org.robolectric.shadows.ShadowContextImpl.getSystemService(ShadowContextImpl.java:81)
	at android.app.ContextImpl.getSystemService(ContextImpl.java)
	at android.content.ContentProvider.attachInfo(ContentProvider.java:2372)
	at android.content.ContentProvider.attachInfo(ContentProvider.java:2358)
	at org.robolectric.android.controller.ContentProviderController.create(ContentProviderController.java:79)
	at org.robolectric.android.controller.ContentProviderController.create(ContentProviderController.java:44)
	at org.robolectric.Robolectric.setupContentProvider(Robolectric.java:61)
	at it.ninniuz.android.robocontentprovider.FailingContentProviderTest.setUp(FailingContentProviderTest.kt:14)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:591)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:274)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:88)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassNotFoundException: android.content.AttributionSource
	at org.robolectric.internal.bytecode.SandboxClassLoader.getByteCode(SandboxClassLoader.java:189)
	at org.robolectric.internal.bytecode.SandboxClassLoader.maybeInstrumentClass(SandboxClassLoader.java:147)
	at org.robolectric.internal.bytecode.SandboxClassLoader.lambda$loadClass$0(SandboxClassLoader.java:133)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:53)
	at org.robolectric.internal.bytecode.SandboxClassLoader.loadClass(SandboxClassLoader.java:133)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2473)
	at org.robolectric.internal.bytecode.ShadowWrangler.findShadowMethodDeclaredOnClass(ShadowWrangler.java:319)
	at org.robolectric.internal.bytecode.ShadowWrangler.findShadowMethod(ShadowWrangler.java:288)
	at org.robolectric.internal.bytecode.ShadowWrangler.pickShadowMethod(ShadowWrangler.java:265)
	at org.robolectric.internal.bytecode.ShadowWrangler.classInitializing(ShadowWrangler.java:160)
	at org.robolectric.internal.bytecode.RobolectricInternals.classInitializing(RobolectricInternals.java:21)
	at android.app.AppOpsManager.<clinit>(AppOpsManager.java)
	at android.app.SystemServiceRegistry$62.$$robo$$android_app_SystemServiceRegistry_62$createService(SystemServiceRegistry.java:795)
	at android.app.SystemServiceRegistry$62.createService(SystemServiceRegistry.java)
	at android.app.SystemServiceRegistry$62.createService(SystemServiceRegistry.java:790)
	at android.app.SystemServiceRegistry$CachedServiceFetcher.$$robo$$android_app_SystemServiceRegistry_CachedServiceFetcher$getService(SystemServiceRegistry.java:1702)
	at android.app.SystemServiceRegistry$CachedServiceFetcher.getService(SystemServiceRegistry.java)
	at android.app.SystemServiceRegistry.$$robo$$android_app_SystemServiceRegistry$getSystemService(SystemServiceRegistry.java:1382)
	at android.app.SystemServiceRegistry.getSystemService(SystemServiceRegistry.java)
	at android.app.ContextImpl.$$robo$$android_app_ContextImpl$getSystemService(ContextImpl.java:1918)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.robolectric.shadows.ShadowContextImpl$_ContextImpl_$$Reflector4.getSystemService(Unknown Source)
	at org.robolectric.shadows.ShadowContextImpl.getSystemService(ShadowContextImpl.java:81)
	at android.app.ContextImpl.getSystemService(ContextImpl.java)
	at android.content.ContentProvider.$$robo$$android_content_ContentProvider$attachInfo(ContentProvider.java:2372)
	at android.content.ContentProvider.attachInfo(ContentProvider.java)
	at android.content.ContentProvider.$$robo$$android_content_ContentProvider$attachInfo(ContentProvider.java:2358)
	at android.content.ContentProvider.attachInfo(ContentProvider.java)
	at org.robolectric.android.controller.ContentProviderController.create(ContentProviderController.java:79)
	at org.robolectric.android.controller.ContentProviderController.create(ContentProviderController.java:44)
	at org.robolectric.Robolectric.setupContentProvider(Robolectric.java:61)
	at it.ninniuz.android.robocontentprovider.FailingContentProviderTest.setUp(FailingContentProviderTest.kt:14)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	... 11 more
```

**Comment by utzcoz** [comment]
@ninniuz Thanks for pointing out this problem. Looks like https://github.com/robolectric/robolectric/blame/1af2c5e1060d53448a42ce6c1d61f4c29f20f7d7/shadows/framework/src/main/java/org/robolectric/shadows/ShadowAppOpsManager.java#L397-L409 uses class `AttributionSource` added from Android S(SDK 31). The `Target29CompatibilityTest` only tests the backward compatibility of `LocationService`.

**Comment by utzcoz** [comment]
@ninniuz could you bump you compileSdk to 31 or use 4.6.1 for your tests temporarily? Maybe this problem needs some time to fix.

**Comment by ninniuz** [comment]
@utzcoz no problem, I will stick with v4.6.1 until I can either bump compileSdk or robolectric. Thanks for your precious help!

**Comment by hoisie** [comment]
This should now be fixed in 4.7.3 (may take a minute or two for that to appear on MavenCentral).
